docs/aws/audit/kmsmonitoring/rules/secret_encrypted_kms_cmk.mdx

```diff
@@ -23,6 +23,281 @@ ISO27001, HIPAA, NISTCSF, PCIDSS
+</Accordion>
+
+<Accordion title='Using CLI'>
+To ensure that Secret Manager secrets are encrypted with Customer Master Keys (CMKs) in AWS KMS using AWS CLI, follow these steps:
+
+1. **Create a Customer Master Key (CMK) in AWS KMS:**
+   First, you need to create a CMK in AWS KMS if you don't already have one. This key will be used to encrypt your secrets.
+
+   ```sh
+   aws kms create-key --description "My CMK for Secret Manager" --key-usage ENCRYPT_DECRYPT --origin AWS_KMS
+   ```
+
+2. **Get the Key ID or ARN of the CMK:**
+   After creating the CMK, you need to retrieve its Key ID or ARN to use it for encryption.
+
+   ```sh
+   aws kms list-keys
+   ```
+
+   This command will list all your keys. Note down the Key ID or ARN of the CMK you created.
+
+3. **Create a Secret with the CMK:**
+   When creating a new secret, specify the CMK to be used for encryption.
+
+   ```sh
+   aws secretsmanager create-secret --name MySecret --secret-string "MySecretValue" --kms-key-id <Your-CMK-Key-ID-or-ARN>
+   ```
+
+4. **Update an Existing Secret to Use the CMK:**
+   If you have an existing secret that is not encrypted with a CMK, you can update it to use the CMK.
+
+   ```sh
+   aws secretsmanager update-secret --secret-id <Your-Secret-ID> --kms-key-id <Your-CMK-Key-ID-or-ARN>
+   ```
+
+By following these steps, you ensure that your secrets in AWS Secret Manager are encrypted using Customer Master Keys (CMKs) in AWS KMS, enhancing the security of your sensitive information.
+</Accordion>
+
+<Accordion title='Using Python'>
+To ensure that Secret Manager secrets are encrypted with Customer Managed Keys (CMKs) in KMS using Python scripts, you can follow these steps for AWS, Azure, and GCP:
+
+### AWS (Amazon Web Services)
+
+1. **Install AWS SDK for Python (Boto3):**
+   ```bash
+   pip install boto3
+   ```
+
+2. **Create a KMS Key:**
+   ```python
+   import boto3
+
+   kms_client = boto3.client('kms')
+   response = kms_client.create_key(
+       Description='My CMK for Secret Manager',
+       KeyUsage='ENCRYPT_DECRYPT',
+       Origin='AWS_KMS'
+   )
+   cmk_id = response['KeyMetadata']['KeyId']
+   print(f"Created CMK with ID: {cmk_id}")
+   ```
+
+3. **Create a Secret with CMK:**
+   ```python
+   import boto3
+
+   secrets_client = boto3.client('secretsmanager')
+   secret_name = "my_secret"
+   secret_value = "my_secret_value"
+
+   response = secrets_client.create_secret(
+       Name=secret_name,
+       SecretString=secret_value,
+       KmsKeyId=cmk_id
+   )
+   print(f"Created secret with ARN: {response['ARN']}")
+   ```
+
+### Azure (Microsoft Azure)
+
+1. **Install Azure SDK for Python:**
+   ```bash
+   pip install azure-identity azure-keyvault-secrets azure-keyvault-keys
+   ```
+
+2. **Create a Key Vault and a Key:**
+   ```python
+   from azure.identity import DefaultAzureCredential
+   from azure.keyvault.keys import KeyClient
+   from azure.keyvault.secrets import SecretClient
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>